crates/ruff_python_parser/src/lexer.rs

```diff
@@ -102,7 +104,14 @@ impl<'src> Lexer<'src> {
 
         // TODO: Handle possible mismatch between BOM and explicit encoding declaration.
         // spell-checker:ignore feff
-        lexer.cursor.eat_char('\u{feff}');
+        let start_offset = if lexer.cursor.eat_char(BOM) {
+            // The start offset needs to consider the BOM character if it's present.
+            start_offset
+                .checked_sub(BOM.text_len())
+                .unwrap_or(start_offset)
+        } else {
+            start_offset
+        };
```

Comment:
I think we can simplify this by only lexing the bom if the start offset is 0. This removes the need for the subtraction (which confused me a bit).

```suggestion
        if start_offset == TextSize::new(0) {
					lexer.cursor_eat_char(BOM);
				} else {
					lexer.cursor.skip_bytes(start_offset.to_usize());
			  }
```